### PR TITLE
Add more api tests for webdav search api

### PR DESCRIPTION
**Pull request by paurakhsharma**
## Description
Add more tests for WebDAV search API.

Related issue: #32602
## How Has This Been Tested?
Locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Acceptance test

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: tests/acceptance/features/apiWebdavOperations/search.feature [hunk 1]
```diff
@@ -8,40 +8,58 @@ Feature: Search
     Given user "user0" has been created
     And user "user0" has created a folder "/just-a-folder"
     And user "user0" has created a folder "/फनी näme"
+    And user "user0" has created a folder "/upload folder"
     And user "user0" has uploaded file with content "does-not-matter" to "/upload.txt"
     And user "user0" has uploaded file with content "does-not-matter" to "/a-image.png"
     And user "user0" has uploaded file with content "does-not-matter" to "/just-a-folder/upload.txt"
+        And user "user0" has uploaded file with content "does-not-matter" to "/just-a-folder/lolol.txt"
```
Comment (individual-it) [inline tests/acceptance/features/apiWebdavOperations/search.feature]: indent

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 2]
```diff
@@ -3034,47 +3034,47 @@ public function userRestoresVersionIndexOfFile($user, $versionIndex, $path) {
 	}
 
 	/**
-	 * @Then /^the (?:propfind|search) result of "([^"]*)" should (not|)\s?contain these files:$/
+	 * @Then /^the (?:propfind|search) result of "([^"]*)" should (not|)\s?contain these entries:$/
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/WebDav.php]: make it more universal, so old steps don't break
```suggestion
	 * @Then /^the (?:propfind|search) result of "([^"]*)" should (not|)\s?contain these (?:files|entries):$/
```

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 3]
```diff
@@ -3034,47 +3034,47 @@ public function userRestoresVersionIndexOfFile($user, $versionIndex, $path) {
...
-					"response does contain the file '$expectedFile[0]' but should not"
+					"response does contain the entry '$expectedFile[0]' but should not"
 				);
 			}
 		}
 	}
 
 	/**
-	 * @Then the propfind/search result of :user should contain :numFiles files
+	 * @Then the propfind/search result of :user should contain :numFiles entries
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/WebDav.php]: make more universal

File: tests/acceptance/features/apiWebdavOperations/search.feature [hunk 4]
```diff
@@ -50,17 +68,59 @@ Feature: Search
...
+  Scenario Outline: limit returned search entries
+    Given using <dav_version> DAV path
+    When user "user0" searches for "upload" and limits the entries to "3" items using the WebDAV API
+    Then the HTTP status code should be "207"
+    And the search result of "user0" should contain "3" entries
+    And the search result of "user0" should contain these entries:
+      | /just-a-folder/upload.txt    |
+      | /just-a-folder/uploadÜठिF.txt |
+      | /upload folder               |
+      | /फनी näme/upload.txt          |
```
Comment (individual-it) [inline tests/acceptance/features/apiWebdavOperations/search.feature]: this are 4 entries
maye add also a "should not contain" section

**Review by individual-it — CHANGES_REQUESTED** [review]
search also for UTF characters in search string

**Review by individual-it — APPROVED** [review]

**Comment by paurakhsharma** [comment]
Backport on: #33547

**Pull request merged**